src/basic/virt.c

```diff
@@ -240,7 +240,7 @@ static bool detect_vm_xen_dom0(void) {
                 return false;
         }
         if (r < 0)
-                return r;
+                return false;
```

Comment:
Hmm, shouldn't this function be changed to return a proper error like all other `detect_vm_*` functions, and the error handled in the caller?

Reply:
This function answers a yes/no question. Errors are of no concern to the caller.

Reply:
well, there are good reasons to propagate unexpected errors... i.e. if we get EPERM on something we shouldn't get it for. I agree too with @keszybz that we should probably propagate errors here, and only demote the expected errors to true/false depending what they indicate

Reply:
Is it a good idea to propagate errors? dmi not accessible: early abort, /sys/hypervisor/ not  accessible: early abort. And so on. Well, I will rewrite `detect_vm_xen_dom0` to propagate errors via int instead of bool.

Reply:
If it's not accessible because -ENOENT, that's fine. If it's -EPERM then we don't know: we might be running on xen, with might not, but all we can say is that most likely there's some borked selinux or seccomp policy. If we cannot give a good answer, it's better to report "unknown" to the caller.